docs/developer/architecture/solutions.rst

```diff
@@ -0,0 +1,241 @@
+
+There is no explicit information about what an archive contains.
+Instead, we want the archive itself to contain such information and to
+discover it at import time.
+
+This configuration file will be read by a Salt external pillar module,
+which will permit the consumption of it by Salt modules and states.
+
+Archive format
+~~~~~~~~~~~~~~
+
+Solution archives must use the `ISO-9660:1988`_ format, including `Rock Ridge`_
+and Joliet_ directory records. The character encoding must be UTF-8_. The
+conformance level is expected to be at most 3, meaning:
+
+- Directory identifiers may not exceed 31 characters (bytes) in length
+- File name + ``'.'`` + file name extension may not exceed 30 characters
+  (bytes) in length
+- Files are allowed to consist of multiple sections
+
+The generated archive should specify a volume ID, set to
+``{project_name} {version}``.
+
+.. _`ISO-9660:1988`: https://www.iso.org/obp/ui/#iso:std:iso:9660:ed-1:v1:en
+.. _`Rock Ridge`: https://en.wikipedia.org/wiki/Rock_Ridge
+.. _Joliet: https://en.wikipedia.org/wiki/Joliet_(file_system)
+.. _UTF-8: https://tools.ietf.org/html/rfc3629
+
+OCI Images registry
+~~~~~~~~~~~~~~~~~~~
+
+Every container images from Solution archive will be exposed as a single
+repository through MetalK8s registry. The name of this repository will be
+computed from the product information ``<NAME>-<VERSION>``.
+
+Solution environment
+~~~~~~~~~~~~~~~~~~~~
+
+Solutions will be deployed into an ``Environment``, which is basically a
+namespace or a group of namespaces with a specific label
+``solutions.metalk8s.scality.com/environment``. It allows to run multiple
+instances of a Solution, with different versions, on the same cluster, without
+collision between them.
+
+Solution management
+~~~~~~~~~~~~~~~~~~~
+
+We will provide CLI and UI to import, unimport and handle the whole lifecycle
+of a Solution. These tools are wrapper around Salt formulas.
+
+Interaction diagram
+~~~~~~~~~~~~~~~~~~~
+
+We include a detailed interaction sequence diagram for describing how MetalK8s
+will handle user input when deploying / upgrading Solutions.
+
+.. uml:: solutions.uml
+
+Rejected design choices
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Solutions vs CNAB
+~~~~~~~~~~~~~~~~~
+
+The Cloud Native Application Bundle (CNAB_) is a standard packaging format
+for multi-component distributed applications. It basically offers what MetalK8s
+Solution does, but with the need of an extra container with almost full access
+to the Kubernetes cluster and that’s the reason why we did choose to not use
+it.
+
+We also want to enforce some practices (Operator, UI, ...) in Solutions
+and this is not easily doable using it.
+
+Moreover, CNAB_ is a pretty young project and has not yet been adopted by a
+lot of people, so it's hard to predict its future.
+
+.. _CNAB: https://cnab.io
+
+Implementation Details
+----------------------
+
+.. todo:: Try to details more / see if there is missing stuff.
+
+Iteration 1
+~~~~~~~~~~~
+
+* Solution example.
```

Comment:
Mention that this example is "fake" - no real application deployed, simply used for testing Metal tooling.